--- usermode/SConstruct	2016-05-25 13:46:56.704338675 -0500
+++ /u/s/a/sankey/Desktop/mnemosyne/usermode/SConstruct	2016-05-17 16:33:47.000000003 -0500
@@ -46,6 +46,11 @@
            type='string',
            metavar='NAME',
            help='a regular expression used to match tests to run')
+AddOption('--enable-trace',
+           action="store_true", dest='enable_trace',
+           default = False,
+           help='Enable tracing of NVM accesses [%default]')
+
 
 # Custom help message. Overrides the help output of the local options.
 Help("""
@@ -77,6 +82,7 @@
 if mainEnv['BUILD_CONFIG_NAME'] is None:
 	mainEnv['BUILD_CONFIG_NAME'] = 'default'
 mainEnv['TEST_FILTER'] = GetOption('test_filter')
+mainEnv['ENABLE_TRACE'] = GetOption('enable_trace')
 
 rootDir = os.getcwd()
 mainEnv['MY_ROOT_DIR'] = rootDir
@@ -85,7 +91,9 @@
 mainEnv['ICC_DIR'] = '/opt/intel/Compiler/11.0/606/'
 
 mainEnv['CC'] = os.path.join(mainEnv['ICC_DIR'], 'bin/intel64/icc')
+#mainEnv['CC'] = os.path.join(mainEnv['ICC_DIR'], 'bin/icc')
 mainEnv['CXX'] = os.path.join(mainEnv['ICC_DIR'],  'bin/intel64/icpc')
+#mainEnv['CXX'] = os.path.join(mainEnv['ICC_DIR'],  'bin/icpc')
 
 
 Export('mainEnv')
@@ -95,11 +103,11 @@
 mcoreLibrary = SConscript('library/mcore/SConscript', 'CommonObjects', variant_dir = 'build/library/mcore')
 mtmLibrary = SConscript('library/mtm/SConscript', variant_dir = 'build/library/mtm')
 
-toolsEnv = mainEnv.Clone()
-toolsEnv['CPPPATH'] = ['#library/mcore/include', '#library/mtm/include']
-Export('toolsEnv')
-Export('mcoreLibrary', 'pmallocLibrary', 'mtmLibrary')
-SConscript('tool/SConscript', variant_dir = os.path.join('build', 'tool'))
+#toolsEnv = mainEnv.Clone()
+#toolsEnv['CPPPATH'] = ['#library/mcore/include', '#library/mtm/include']
+#Export('toolsEnv')
+#Export('mcoreLibrary', 'pmallocLibrary', 'mtmLibrary')
+#SConscript('tool/SConscript', variant_dir = os.path.join('build', 'tool'))
 
 # Explicit dependencies that were built outside SCons
 mainEnv['EXPLICIT_DEPENDENCY'] = {}
